### PR TITLE
Fixes random gear sometimes not spawning

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Title

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!bugfix

File: code/controllers/subsystems/job.dm [hunk 1]
```diff
@@ -813,7 +823,12 @@
 			for (var/thing in items)
 				to_chat(H, "<span class='notice'>Placing \the [thing] in your [B.name]!</span>")
 				var/datum/gear/G = gear_datums[thing]
-				var/metadata = prefs.gear[G.display_name]
+				var/metadata
+				var/list/gear_test = prefs.gear[G.display_name]
+				if(gear_test.len)
+					metadata = gear_test
+				else
+					metadata = G.display_name
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]:
```suggestion
					metadata = list()
```

File: code/controllers/subsystems/job.dm [hunk 2]
```diff
@@ -813,7 +823,12 @@
 			for (var/thing in items)
 				to_chat(H, "<span class='notice'>Placing \the [thing] in your [B.name]!</span>")
 				var/datum/gear/G = gear_datums[thing]
-				var/metadata = prefs.gear[G.display_name]
+				var/metadata
+				var/list/gear_test = prefs.gear[G.display_name]
+				if(gear_test.len)
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]:
```suggestion
				if(gear_test?.len)
```

File: code/controllers/subsystems/job.dm [hunk 3]
```diff
@@ -787,7 +792,12 @@
 		if (G.slot in used_slots)
 			. += thing
 		else
-			var/metadata = prefs.gear[G.display_name]
+			var/metadata
+			var/list/gear_test = prefs.gear[G.display_name]
+			if(gear_test.len)
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]:
```suggestion
			if(gear_test?.len)
```

File: code/controllers/subsystems/job.dm [hunk 4]
```diff
@@ -787,7 +792,12 @@
...
+				metadata = gear_test
+			else
+				metadata = G.display_name
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]:
```suggestion
				metadata = list()
```

File: code/controllers/subsystems/job.dm [hunk 5]
```diff
@@ -754,7 +754,12 @@
 			if(G.slot && !(G.slot in custom_equip_slots))
 				// This is a miserable way to fix the loadout overwrite bug, but the alternative requires
 				// adding an arg to a bunch of different procs. Will look into it after this merge. ~ Z
-				var/metadata = prefs.gear[G.display_name]
+				var/metadata
+				var/list/gear_test = prefs.gear[G.display_name]
+				if(gear_test.len)
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]:
```suggestion
				if(gear_test?.len)
```

File: code/controllers/subsystems/job.dm [hunk 6]
```diff
@@ -754,7 +754,12 @@
...
+					metadata = gear_test
+				else
+					metadata = G.display_name
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]:
```suggestion
					metadata = list()
```

**Comment by NonQueueingMatt** [comment]
wayment

**Pull request merged**